File: SWB4/Client/OpenOffice/OfficeComunicationInterfaces/src/org/semanticwb/office/interfaces/CategoryInfo.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package org.semanticwb.office.interfaces;

/**
 *
 * @author victor.lorenzana
 */
public class CategoryInfo {    
    public String UDDI;
    public String title;
    public String description;
    @Override
    public String toString()
    {

        return title;
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == null)
        {
            return false;
        }
        if (getClass() != obj.getClass())
        {
            return false;
        }
        final CategoryInfo other = (CategoryInfo) obj;
        if ((this.UDDI == null) ? (other.UDDI != null) : !this.UDDI.equals(other.UDDI))
        {
            return false;
        }
        return true;
    }

    @Override
    public int hashCode()
    {
        int hash = 3;
        hash = 97 * hash + (this.UDDI != null ? this.UDDI.hashCode() : 0);
        return hash;
    }
}

File: SWB4/Client/Microsoft Office/INFOTEC WebBuilder 4/WBOffice4/Interfaces/CategoryInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace WBOffice4.Interfaces
{
    public class CategoryInfo
    {
        public String UDDI;
        public String title;
        public String description;
        public int childs;
        public override string ToString()
        {
            return title;
        }
        public override int GetHashCode()
        {
            return UDDI.GetHashCode();
        }
        public override bool Equals(Object obj)
        {
            bool equals = false;
            if (obj is CategoryInfo)
            {
                return ((CategoryInfo)obj).UDDI == UDDI;
            }
            return equals;
        }

    }
}